### ISSUE TITLE
Please build Syncthing with GOARM

### ISSUE THREAD

**Issue by drizzt** [syncthing/syncthing#5305]
Hi,
I see that syncthing is built, for arm, without specify the [GOARM variable](https://github.com/golang/go/wiki/GoArm).

I tested on my NAS (Synology DS416j) and I see a big difference from the official build (linux-arm) and a custom build, built with GOARM=7.

Since the time to build Syncthing is very low I suggest you to change build.go and the release script in order to build for arm5 (GOARM=5), arm6 (GOARM=6) and arm7 (GOARM=7) as different builds

**Comment by calmh** [comment]
Our binaries are built with GOARM=5. The upgrade system looks for a binary matching GOOS-GOARCH, and these do not differ between GOARM variants, so we can only easily provide one binary version. We used to do it differently, but it leads to all kinds of interesting issues, especially for people who actually build the binary *on* ARM, and we stopped doing that (#1051, #1218). I'm content with providing just a single lowest-common-denominator binary for 32 bit arm for now.
